packages/status-js/src/client/community/community.ts

```diff
@@ -1,82 +1,66 @@
-import { hexToBytes } from 'ethereum-cryptography/utils'
-import { PageDirection, waku_message } from 'js-waku'
-import difference from 'lodash/difference'
+import { waku_message } from 'js-waku'
 
-import { ChatMessage } from '~/protos/chat-message'
-import { CommunityRequestToJoin } from '~/protos/communities'
-import { EmojiReaction } from '~/protos/emoji-reaction'
+import { MessageType } from '~/protos/enums'
+import { getDifferenceByKeys } from '~/src/helpers/get-difference-by-keys'
 
 import { idToContentTopic } from '../../contentTopic'
 import { createSymKeyFromPassword } from '../../encryption'
-import { createChannelContentTopics } from './create-channel-content-topics'
+import { Chat } from '../chat'
 
 import type { Client } from '../../client'
-import type { CommunityDescription } from '../../wire/community_description'
-import type { Reactions } from './get-reactions'
-import type { ImageMessage } from '~/src/proto/communities/v1/chat_message'
-import type { Waku } from 'js-waku'
-
-export type CommunityMetadataType = CommunityDescription['proto']
-
-export type MessageType = ChatMessage & {
-  messageId: string
-  pinned: boolean
-  reactions: Reactions
-  channelId: string
-  responseToMessage?: Omit<MessageType, 'responseToMessage'>
-}
+import type {
+  CommunityChat,
+  CommunityDescription,
+} from '~/src/proto/communities/v1/communities'
 
 export class Community {
   private client: Client
-  private waku: Waku
-  public communityPublicKey: string
-  private communityContentTopic!: string
-  private communityDecryptionKey!: Uint8Array
-  public communityMetadata!: CommunityMetadataType // state
-  public channelMessages: Partial<{ [key: string]: MessageType[] }> = {} // state
-  public channelMessagesCallbacks: {
-    [key: string]: (messages: MessageType[]) => void
-  } = {}
-  public communityCallback:
-    | ((community: CommunityMetadataType) => void)
-    | undefined
-
-  constructor(client: Client, waku: Waku, publicKey: string) {
+
+  public publicKey: string
+  private contentTopic!: string
+  private symetricKey!: Uint8Array
+  // todo?: rename to community (and access via commuity.community)
+  public description!: CommunityDescription
+  public chats: Map<string, Chat>
```

Comment:
Notice: chats as prop now

Reply:
Use public methods intead.

Reply:
Await https://github.com/status-im/status-web/pull/270#discussion_r895414850.

Then turn private and add public `getChat(chatUuid: string)` or `getChats()`.

Reply:
We'll use a getter instead.